libraries/AP_HAL_ChibiOS/hwdef/NxtPX4v2/hwdef.dat

```diff
@@ -0,0 +1,164 @@
+# hw definition file for processing by chibios_hwdef.py
+# for the NxtPX4v2 hardware
+
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1157
+
+# crystal frequency
+OSCILLATOR_HZ 16000000
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 12
+define CH_CFG_ST_RESOLUTION 16
+
+# flash size
+FLASH_SIZE_KB 2048
+FLASH_RESERVE_START_KB 128
+
+define HAL_STORAGE_SIZE 32768
+STORAGE_FLASH_PAGE 14
+
+# default to all pins low to avoid ESD issues
+DEFAULTGPIO OUTPUT LOW PULLDOWN
+
+# order of UARTs (and USB)
+SERIAL_ORDER  OTG1 USART2 UART4 USART1 USART3 UART7 UART5 UART8
+
+# USB
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+PD0 VBUS INPUT OPENDRAIN
+
+# GPS1
+PA9  USART1_TX USART1
+PA10 USART1_RX USART1
+
+# TELEM1
+PD6 USART2_RX USART2
+PD5 USART2_TX USART2
+
+# DJI O3
+PD8 USART3_TX USART3
+PD9 USART3_RX USART3
+
+# TELEM2
+PB9 UART4_TX UART4
+PB8 UART4_RX UART4
+
+# RC_INPUT
+PB13 UART5_TX UART5
+PB12 UART5_RX UART5
+PD14 SBUS_INV OUTPUT LOW
+
+# ESC
+PE7 UART7_RX UART7
+
+# TELEM3
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+
+# SWD
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# two I2C bus
+I2C_ORDER I2C1 I2C4
+
+# I2C1 - baro
+PB6 I2C1_SCL I2C1
+PB7 I2C1_SDA I2C1
+
+# I2C4 - CONNECTOR
+PD12 I2C4_SCL I2C4
+PD13 I2C4_SDA I2C4
+
+# PWM output pins
+PE9  TIM1_CH1 TIM1  PWM(1) GPIO(50)
```

Comment:
It's an H7, you should provide bi-dir outputs at least for the first 4 motors, but possibly 8

Reply:
Okay! I have enable BDshot and tested it in my drone. It works well.